### ISSUE TITLE
关于 fluid.layers.conv2d 中的padding参数

### ISSUE THREAD

**Issue by supyer9** [PaddlePaddle/Paddle#26900]
您好，我想请教下，fluid.layers.conv2d 中 padding=0 和 padding = n 对应于 tensorflow 中 tf.nn.conv2d 中的 padding 要怎么设置呢？
我目前的做法是，对于 paddle 中的padding=n 。 在tensorflow中， 先用 tf.pad 在 input 周围 pad 上 n 个0 ，之后设置 tf.nn.conv2d参数padding=VALID, 请问这样的做法是正确的吗？

**Comment by cryoco** [comment]
您好，paddle和tf的conv2d都支持通过padding=一个包含4个二元组的list来显式设置padding，您可以通过这种方式来对应到tf。
文档：
[paddle conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/conv2d_cn.html#conv2d)
[tf conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d?hl=en)

**Comment by supyer9** [comment]
> 您好，paddle和tf的conv2d都支持通过padding=一个包含4个二元组的list来显式设置padding，您可以通过这种方式来对应到tf。
> 文档：
> [paddle conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/conv2d_cn.html#conv2d)
> [tf conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d?hl=en)

感谢您的回复，tensorflow 1.+ 版本中好像并不支持 padding=list 的操作，明确要求padding='str' ，即 ‘SAME’或‘VALID’
在这种限制条件下，我问题中提到的解决方法，是否可行呢？谢谢

**Comment by cryoco** [comment]
> > 您好，paddle和tf的conv2d都支持通过padding=一个包含4个二元组的list来显式设置padding，您可以通过这种方式来对应到tf。
> > 文档：
> > [paddle conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/conv2d_cn.html#conv2d)
> > [tf conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d?hl=en)
> 
> 感谢您的回复，tensorflow 1.+ 版本中好像并不支持 padding=list 的操作，明确要求padding='str' ，即 ‘SAME’或‘VALID’
> 在这种限制条件下，我问题中提到的解决方法，是否可行呢？谢谢

设定same或valid时，输出的HW与stride或dilation有关，相关公式：

**Comment by cryoco** [comment]
> > 您好，paddle和tf的conv2d都支持通过padding=一个包含4个二元组的list来显式设置padding，您可以通过这种方式来对应到tf。
> > 文档：
> > [paddle conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/conv2d_cn.html#conv2d)
> > [tf conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d?hl=en)
> 
> 感谢您的回复，tensorflow 1.+ 版本中好像并不支持 padding=list 的操作，明确要求padding='str' ，即 ‘SAME’或‘VALID’
> 在这种限制条件下，我问题中提到的解决方法，是否可行呢？谢谢

设定为same或valid时，输出HW与stride或dilation有关，相关公式可以参考
![image](https://user-images.githubusercontent.com/12407750/91957981-53055a80-ed39-11ea-984b-8d05ab76b0ef.png)
可以看下你的使用方式是否可以通过设置stride或dilation对应到这两种情况，如果没有办法对应，可能需要更新tf到2.0+了

**Comment by supyer9** [comment]
> > > 您好，paddle和tf的conv2d都支持通过padding=一个包含4个二元组的list来显式设置padding，您可以通过这种方式来对应到tf。
> > > 文档：
> > > [paddle conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/conv2d_cn.html#conv2d)
> > > [tf conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d?hl=en)
> > 
> > 
> > 感谢您的回复，tensorflow 1.+ 版本中好像并不支持 padding=list 的操作，明确要求padding='str' ，即 ‘SAME’或‘VALID’
> > 在这种限制条件下，我问题中提到的解决方法，是否可行呢？谢谢
> 
> 设定为same或valid时，输出HW与stride或dilation有关，相关公式可以参考
> ![image](https://user-images.githubusercontent.com/12407750/91957981-53055a80-ed39-11ea-984b-8d05ab76b0ef.png)
> 可以看下你的使用方式是否可以通过设置stride或dilation对应到这两种情况，如果没有办法对应，可能需要更新tf到2.0+了

好的，谢谢回复
